src/cplus_plugin/gui/qgis_cplus_main.py

```diff
@@ -1013,90 +1024,235 @@ def open_help(self):
+                metadata_id,
+                current_view,
+                constant_raster_dialog,
+            )
+        )
+
+        constant_raster_dialog.exec_()
+
+    def _on_constant_rasters_create_requested(
+        self,
+        context,
+        collection,
+        input_range: tuple,
+        metadata_id: str,
+        current_view: bool = True,
+        dialog=None,
+    ):
+        """Handle constant raster creation request from the dialog.
+
+        :param context: ConstantRasterContext with extent, CRS, output directory
+        :param collection: ConstantRasterCollection to create rasters from
+        :param input_range: Tuple of (min, max) for input values
+        :param metadata_id: Metadata identifier for the constant raster type
+        :param current_view: If True, create for current view only; if False, create for all types
+        :param dialog: Reference to the dialog for emitting completion signals
+        """
+        # Create progress dialog
+        progress_dialog = QtWidgets.QProgressDialog(
+            "Creating constant rasters...", "Cancel", 0, 100, self
+        )
+        progress_dialog.setWindowTitle("Creating Constant Rasters")
+        progress_dialog.setWindowModality(QtCore.Qt.WindowModal)
+        progress_dialog.show()
+
+        # Create feedback
+        feedback = QgsProcessingFeedback()
+
+        # Connect feedback to progress dialog
+        def update_progress(progress):
+            progress_dialog.setValue(int(progress))
+
+        def update_label(message):
+            progress_dialog.setLabelText(message)
+
+        feedback.progressChanged.connect(update_progress)
+        feedback.pushInfo = lambda msg: update_label(msg)
+
+        try:
+            if current_view:
+                # Create rasters for current view only
+                created_rasters = ConstantRasterProcessingUtils.create_constant_rasters(
+                    collection, context, input_range, feedback, metadata_id
+                )
+
+                progress_dialog.close()
+
+                if created_rasters:
+                    message = f"Created {len(created_rasters)} constant raster(s) in {context.output_dir}"
+                    if dialog:
+                        dialog.raster_creation_completed.emit(
+                            True, message, len(created_rasters)
+                        )
+                    else:
+                        self.show_message(message, level=Qgis.Success)
+                else:
+                    message = "No enabled components found in the collection."
+                    if dialog:
+                        dialog.raster_creation_completed.emit(False, message, 0)
+                    else:
+                        self.show_message(message, level=Qgis.Warning)
+            else:
+                # Create rasters for all constant raster types
+                all_created_rasters = []
+                metadata_ids = constant_raster_registry.metadata_ids()
```

Comment:
@osundwajeff constant raster registry is iterable since it implements `__iter__` hence it might be slightly faster to:
```
for metadata in constant_raster_registry:
   collection = metadata.collection
```